### ISSUE TITLE
Is it possible to embed or inherit all fields from one Structure into another?

### ISSUE THREAD

**Issue by jabalsad** [awslabs/smithy#883]
Hi,

I'm trying to reuse structures using some form of embedding or inheritance, in the same way the aggregate shapes Lists and Maps use members to point to Structures.

E.g.

```
structure Foo {
  bar: String
}

structure Child {
  member: Foo
  somethingElse: String
}
```

But I've noticed these get directly translated to objects that have a field named `member` instead of inheriting the fields from the other structure. Is this at all possible with the Smithy IDL?

**Comment by kstich** [comment]
Not currently. There is an accepted proposal for mixins available for an upcoming version of the IDL [that you can view here](https://github.com/awslabs/smithy/blob/idl-1.1/designs/mixins.md). Your example would translate to this:

```
@mixin
structure Foo {
    bar: String
}

structure Child with Foo {
    somethingElse: String
}
```

This would be equivalent to the flattened structure of:

```
structure Child {
    bar: String,
    somethingElse: String
}
```

**Comment by jabalsad** [comment]
That would be perfect. I support this proposal! :)

**Comment by DanielBauman88** [comment]
Mixins doesn't address some use cases which are important. Please correct me if there's something I'm missing.

Lets say I'm building a service for customers to generate random numbers. I define RandomNumberService in smithy.
All my errors thrown from RandomNumberService have error messages and error codes.
The error codes are enums which customers can use for their control logic. The messages are descriptive text.

```
@error("client")
@httpError(429)
structure RateLimitExceededError {
  message: string,
  code: ErrorCode (an enum, one value would be RATE_LIMIT_EXCEEDED)
}
```

So customers now can't have any generic logic which handles all service exceptions and logs the code. This would be the case even if we could mixin the code into all our error types. Because on the client side there would be no base exception with a code to handle in a generic way.

```
    if (e instanceof RandomNumberServiceException) {
      console.log(
        `Got error calling RandomNumber Service. Code: ${e.code}. Message: ${e.message} RequestId: ${e.$metadata.requestId}`
      );
    }
```

This is essentialy the pattern AWS implements for SDK errors and the way errors can be handled in sdk clients. Is there a way to achieve the same with smithy as it is today?

**Comment by mtdowling** [comment]
Mixins were released yesterday as part of Smithy IDL 2.0 (1.23 release): https://aws.amazon.com/blogs/developer/introducing-smithy-idl-2-0/
